proxies/live/apiproxy/targets/apim.xml

```diff
@@ -16,18 +19,10 @@
         <Name>RaiseFault.400BadRequest</Name>
         <Condition>request.verb != "GET" and request.header.content-type != "application/fhir+json"</Condition>
       </Step>
-      <!--Need full requirements to implement-->
-      <!-- <Step>
-        <Name>AssignMessage.PopulateAsidFromApp</Name>
-          <Condition>(app.asid IsNot null)</Condition>
-      </Step>
       <Step>
         <Name>RaiseFault.400BadRequest</Name>
-          <Condition>(app.asid is null)</Condition>
+          <Condition>(verifyapikey.VerifyAPIKey.CustomAttributes.asid is null) or (verifyapikey.VerifyAPIKey.CustomAttributes.ods is null)</Condition>
```

Comment:
Rather than returning a 400 should we return a 500? Because the error is at our side not the consumer. Also we need to follow the standards from this confluence [page](https://nhsd-confluence.digital.nhs.uk/display/APM/Handling+Errors#HandlingErrors-1.Createa%22CatchAll%22ErrorMessage).